### PR TITLE
Issue/7710 whats next progress

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Add a ***What's Next*** checklist to show the progress of tasks as described in https://github.com/wordpress-mobile/WordPress-Android/issues/7710.

There are multiple differences in the interface since the original design. See the descriptions and screenshots below for details.

- What's Next was changed to Quick Start
- Icon color updated to match Sites and have higher contrast for better accessibility
- Completed view layout and text updated
- Skip All button shows confirmation dialog
- Copy updated after Editorial review

--Edit from Klym--
This is just a dummy checklist that I'm going to build individual tutorials on. 
To test:

1. Open `Quick Start` activity from the My Site tab.
2. Make sure "Create your site" task is crossed out by default.
3. Click on different tasks, make sure they get crossed out.
4.  (1) Complete all tasks - notice that "Congrats" view at the top appears and Skip All view at the bottom is gone (this is temp behavior, and scroll view does not have to scroll to the top to make "Congrats" view visible).
4. (2) Press Skip All view at the bottom, skip all tasks and make sure that "Congrats" view at the top appears and Skip All view at the bottom is gone.

list_progress

list_complete

list_skip

### CODE REVIEW

**Comment by khaykov** [comment]
Part of #7707

**Comment by khaykov** [comment]
@planarvoid Thanks for taking a look! You are right, the original intention was just to use it as activity, without bottom nav, but it's probably better just to convert it to Fragment, in case we'll change our mind someday. 

I didn't get a chance to utilize `ViewModel` yet, but I'll take a look into it!

**Comment by khaykov** [comment]
@planarvoid I moved things to fragment and `ViewModel`, and added some tests for it. It's my first take on `ViewModel` patter, so I referenced some code from Activity Log. Do let me know if I missed something :)

File: WordPress/src/main/java/org/wordpress/android/viewmodel/quickstart/QuickStartViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import android.arch.lifecycle.ViewModel
+import org.wordpress.android.fluxc.store.QuickStartStore
+import org.wordpress.android.fluxc.store.QuickStartStore.QuickStartTask
+import org.wordpress.android.ui.quickstart.QuickStartTaskState
+import javax.inject.Inject
+
+class QuickStartViewModel @Inject constructor(private val quickStartStore: QuickStartStore) : ViewModel() {
+    private val _quickStartTaskStateStates = MutableLiveData<List<QuickStartTaskState>>()
+    val quickStartTaskStateStates: LiveData<List<QuickStartTaskState>>
+        get() = _quickStartTaskStateStates
```
Comment (planarvoid) [inline WordPress/src/main/java/org/wordpress/android/viewmodel/quickstart/QuickStartViewModel.kt]: NIT: you could do this instead `val quickStartTaskStateStates: LiveData<List<QuickStartTaskState>> = _quickStartTaskStateStates

File: WordPress/src/main/res/layout/my_site_fragment.xml [hunk 2]
```diff
@@ -133,7 +133,7 @@
             <!--Quick Start-->
             <LinearLayout
                 android:id="@+id/row_quick_start"
-                android:visibility="gone"
+                android:visibility="visible"
```
Comment (planarvoid) [inline WordPress/src/main/res/layout/my_site_fragment.xml]: it should be `visible` by default, right? so you can remove this line

File: WordPress/src/main/res/layout/quick_start_activity.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <!--<include-->
```
Comment (planarvoid) [inline WordPress/src/main/res/layout/quick_start_activity.xml]: maybe remove this commented out code?

**Comment by planarvoid** [comment]
The dialog doesn't look the way it should according to the designs. Is it because of the `BasicDialogFragment`?

screenshot_1530192838

File: WordPress/src/main/res/layout/my_site_fragment.xml (hunk 2 above)
Comment (khaykov) [reply WordPress/src/main/res/layout/my_site_fragment.xml]: No, visibility is conditioned (not visible for the majority of users). I made it explicitly visible for testing purposes.

**Comment by khaykov** [comment]
@planarvoid Thanks for taking a look! 
I'm not sure myself, but I assumed `BasicDialogFragment` was a way to deal with dialogs persistent. I changed it to simpler `AlertDialog`, so it will not depend on activity and look as envisioned :)

**Comment by loremattei** [comment]
Hey! I moved the milestone to 10.5 since 10.4 has been cut. Please, let me know it has to land 10.4 so that we can arrange to update the build!

**Pull request merged**